core/primitives/sector_file/sector_file.cpp

```diff
@@ -454,6 +459,38 @@ namespace fc::primitives::sector_file {
     };
   }
 
+  outcome::result<boost::optional<PieceInfo>> SectorFile::writeNull(
+      PaddedByteIndex offset,
+      PaddedPieceSize size,
+      const boost::optional<RegisteredSealProof> &maybe_seal_proof_type) {
+    if (not file_.good()) {
+      return SectorFileError::kInvalidFile;
+    }
+
+    const auto piece = runsAnd(runs_, std::vector<uint64_t>({offset, size}));
+    OUTCOME_TRY(allocated_size, runsCount(piece));
+
+    assert(allocated_size == 0 && "Overwriting with zeros is not available");
+
+    file_.seekp(gsl::narrow<int64_t>(offset), std::ios_base::beg);
+
+    if (not file_.good()) {
+      return SectorFileError::kCannotMoveCursor;
+    }
+
+    OUTCOME_TRY(markAllocated(offset, size));
+
+    if (not maybe_seal_proof_type.has_value()) {
+      return boost::none;
+    }
```

Comment:
check argument earlier, eg immediately after file good check

Reply:
We need to mark allocated